homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,310 @@
+                                     station[ATTR_LATITUDE],
+                                     station[ATTR_LONGITUDE])
+            station_id = station[ATTR_ID]
+            station_uid = str(station.get(ATTR_EXTRA, {}).get(ATTR_UID, ''))
+
+            if radius > dist or stations_list.intersection((station_id,
+                                                            station_uid)):
+                entities.append(CityBikesStation(network, station_id, name))
+
+        async_add_entities(entities, True)
+
+    hass.async_add_job(async_setup_network)
+
+
+class CityBikesNetwork:
+    """Thin wrapper around a CityBikes network object."""
+
+    @classmethod
+    @asyncio.coroutine
+    def get_closest_network_id(cls, hass, latitude, longitude):
+        """Return the id of the network closest to provided location."""
+        try:
+            session = async_get_clientsession(hass)
+            with async_timeout.timeout(5, loop=hass.loop):
+                req = yield from session.get(DEFAULT_ENDPOINT.format(
+                    uri=NETWORKS_URI))
+            json_response = yield from req.json()
+            networks = NETWORKS_RESPONSE_SCHEMA(json_response)
+            network = networks[ATTR_NETWORKS_LIST][0]
+            result = network[ATTR_ID]
+            minimum_dist = location.distance(
+                latitude, longitude,
+                network[ATTR_LOCATION][ATTR_LATITUDE],
+                network[ATTR_LOCATION][ATTR_LONGITUDE])
+            for network in networks[ATTR_NETWORKS_LIST][1:]:
+                network_latitude = network[ATTR_LOCATION][ATTR_LATITUDE]
+                network_longitude = network[ATTR_LOCATION][ATTR_LONGITUDE]
+                dist = location.distance(latitude, longitude,
+                                         network_latitude, network_longitude)
+                if dist < minimum_dist:
+                    minimum_dist = dist
+                    result = network[ATTR_ID]
+
+            return result
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Could not connect to CityBikes API endpoint")
+            return None
+        except ValueError:
+            _LOGGER.error("Received non-JSON data from CityBikes"
+                          " API endpoint")
+            return None
+        except vol.Invalid as err:
+            _LOGGER.error("Received unexpected JSON from CityBikes"
+                          " API endpoint: %s", err)
+            return None
+
+    def __init__(self, hass, network_id):
+        """Initialize the network object."""
+        self._hass = hass
+        self._network_id = network_id
+        self._stations = []
+        self._ready = False
+
+    @property
+    def uid(self):
+        """Return the network ID."""
+        return self._network_id
+
+    @property
+    def stations(self):
+        """Return the stations of the network."""
+        return self._stations
+
+    @property
+    def ready(self):
```

Comment:
You're the only consumer of this object, there is no really need to wrap properties around the instance variables for `uid`, `stations` and `ready`.